tfx/components/infra_validator/model_server_runners/kubernetes_runner.py

```diff
@@ -210,6 +211,30 @@ def _BuildPodManifest(self) -> k8s_client.V1Pod:
           model_path=self._model_path)
       env_vars = [k8s_client.V1EnvVar(name=key, value=value)
                   for key, value in env_vars_dict.items()]
+
+      if self._serving_pod_manifest_overrides and \
+              self._serving_pod_manifest_overrides.env:
+        additional_env_vars = []
+        for env in self._serving_pod_manifest_overrides.env:
+          if env.value:
+            logging.info(f'name={env.name} value={env.value}')
+            additional_env_vars.append(
+              k8s_client.V1EnvVar(name=env.name, value=env.value))
+          else:
```

Comment:
nit: Let's guard this branch by `elif env.valueFrom and env.valueFrom.secretKeyRef:`